### PR TITLE
Make jax.numpy.broadcast_to consistent with numpy.

### PR DESCRIPTION
**Pull request by petebu**
Repro:
In [1]: import jax.numpy as jnp
In [2]: jnp.broadcast_to(10.0, [])
Out[2]: 10.0
In [3]: jnp.broadcast_to(10.0, [1])
Out[3]: DeviceArray([10.], dtype=float32)
In [4]: import numpy as np
In [5]: np.broadcast_to(10.0, [])
Out[5]: array(10.)
In [6]: np.broadcast_to(10.0, [1])
Out[6]: array([10.])

### CODE REVIEW

File: tests/lax_numpy_test.py [hunk 1]
```diff
@@ -2393,6 +2393,9 @@ def testBroadcastToIntIssue1548(self):
     self.assertAllClose(lnp.broadcast_to(1, (3, 2)), onp.ones((3, 2)),
                         check_dtypes=False)
 
+  def testBroadcastToOnScalar(self):
+    self.assertTrue(isinstance(lnp.broadcast_to(10.0, ()), lnp.ndarray))
```
Comment (mattjj) [inline tests/lax_numpy_test.py]: `self.assertIsInstance`?

Comment (mattjj) [reply tests/lax_numpy_test.py]: WDYT about also adding a line like

```python
self.assertIsInstance(onp.broadcast_to(10.0, ()), onp.ndarray))
```

in the spirit of self-documenting code?

**Review by mattjj — APPROVED** [review]
Thanks! Couple ideas below, but LGTM.

File: tests/lax_numpy_test.py (hunk 1 above)
Comment (petebu) [reply tests/lax_numpy_test.py]: Done.

**Pull request merged**